controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/CommitTransactionTask.java

```diff
@@ -0,0 +1,174 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.server.eventProcessor.requesthandlers;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import io.pravega.common.concurrent.Futures;
+import io.pravega.controller.store.stream.OperationContext;
+import io.pravega.controller.store.stream.StoreException;
+import io.pravega.controller.store.stream.StreamMetadataStore;
+import io.pravega.controller.store.stream.TxnStatus;
+import io.pravega.controller.task.Stream.StreamMetadataTasks;
+import io.pravega.shared.controller.event.CommitEvent;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.stream.Collectors;
+
+/**
+ * Request handler for processing commit events in request-stream.
+ */
+@Slf4j
+public class CommitTransactionTask implements StreamTask<CommitEvent> {
+    private final StreamMetadataTasks streamMetadataTasks;
+    private final StreamMetadataStore streamMetadataStore;
+    private final ScheduledExecutorService executor;
+    private final BlockingQueue<CommitEvent> processedEvents;
+
+    @VisibleForTesting
+    public CommitTransactionTask(final StreamMetadataStore streamMetadataStore,
+                                final StreamMetadataTasks streamMetadataTasks,
+                                final ScheduledExecutorService executor,
+                                final BlockingQueue<CommitEvent> queue) {
+        this.streamMetadataStore = streamMetadataStore;
```

Comment:
1. Please rearrange these Constructor's arguments to be in the same order.
2. Have the one with fewer args invoke the one with more args. There is no need to duplicate code, since this will lead to problems like missing null-checks (such as in this method).

Reply:
done